src/languageHandler.js

```diff
@@ -64,75 +61,112 @@ export function _t(...args) {
+    const regexpMapping = {};
+
+    if(variables !== undefined) {
+        for (const variable in variables) {
+            regexpMapping[`%\\(${variable}\\)s`] = variables[variable];
         }
-        if (!(subs[i] instanceof Function)) {
-            throw new Error(`_tJsx: programmer error. expected Function for text: ${jsxText}`);
+    }
+
+    if(tags !== undefined) {
+        for (const tag in tags) {
+            regexpMapping[`(<${tag}>(.*?)<\\/${tag}>|<${tag}>|<${tag}\\s*\\/>)`] = tags[tag];
         }
     }
+    return replaceByRegexes(text, regexpMapping);
+}
 
-    // The translation returns text so there's no XSS vector here (no unsafe HTML, no code execution)
-    const tJsxText = _t(jsxText, {interpolate: false});
-    const output = [tJsxText];
+/*
+ * Replace parts of a text using regular expressions
+ * @param {string} text The text on which to perform substitutions
+ * @param {object} mapping A mapping from regular expressions in string form to replacement string or a
+ * function which will receive as the argument the capture groups defined in the regexp. E.g.
+ * { 'Hello (.?) World': (sub) => sub.toUpperCase() }
+ *
+ * @return a React <span> component if any non-strings were used in substitutions, otherwise a string
+ */
+export function replaceByRegexes(text, mapping) {
+    const output = [text];
+
+    let wrap = false; // Remember if the output needs to be wrapped later
+    for (const regexpString in mapping) {
+        const regexp = new RegExp(regexpString);
 
-    for (let i = 0; i < patterns.length; i++) {
         // convert the last element in 'output' into 3 elements (pre-text, sub function, post-text).
         // Rinse and repeat for other patterns (using post-text).
         const inputText = output.pop();
-        const match = inputText.match(patterns[i]);
-        if (!match) {
-            throw new Error(`_tJsx: translator error. expected translation to match regexp: ${patterns[i]}`);
+        const match = inputText.match(regexp);
+        if(!match) {
+            output.push(inputText); // Push back input
+            continue; // Missing matches is entirely possible, because translation might change things
         }
-        const capturedGroups = match.slice(1);
+        const capturedGroups = match.slice(2);
 
         // Return the raw translation before the *match* followed by the return value of sub() followed
         // by the raw translation after the *match* (not captured group).
         output.push(inputText.substr(0, match.index));
-        output.push(subs[i].apply(null, capturedGroups));
+
+        let toPush;
+        // If substitution is a function, call it
+        if(mapping[regexpString] instanceof Function) {
+            toPush = mapping[regexpString].apply(null, capturedGroups);
+        } else {
+            toPush = mapping[regexpString];
+        }
+
+        output.push(toPush);
+
+        // Check if we need to wrap the output into a span at the end
```

Comment:
I would change this comment to be along the lines of "wrap the output in a React component if any items are React components"